### PR TITLE
[JSX] Remove `hasError` Boolean from Form Elements

### PR DESCRIPTION
**Pull request by HenriRabalais**
## Description

This pull request removes the `hasError` boolean prop from various form elements and updates the error handling logic to rely solely on the `errorMessage` prop. This change simplifies the error handling mechanism and ensures consistency across all form components.

### Changes Made

- **Removed `hasError` prop** from form elements in `jsx/Form.js`, `jsx/Form.d.ts`, and other related files.
- **Updated form components** to check for the presence of `errorMessage` instead of `hasError` to determine if an error state should be displayed.
- **Modified default props and prop types** to remove `hasError` and ensure `errorMessage` is used appropriately.
- **Adjusted form usage** in various modules to reflect this change, including:

  - `battery_manager/jsx/batteryManagerForm.js`
  - `conflict_resolver/jsx/fix_conflict_form.js`
  - `data_release/jsx/addPermissionForm.js`
  - `data_release/jsx/managePermissionsForm.js`
  - `data_release/jsx/uploadFileForm.js`
  - `document_repository/jsx/categoryForm.js`
  - `document_repository/jsx/deleteCategoryForm.js`
  - `document_repository/jsx/editCategoryForm.js`
  - `document_repository/jsx/editForm.js`
  - `document_repository/jsx/uploadForm.js`
  - `electrophysiology_uploader/jsx/UploadForm.js`
  - `imaging_uploader/jsx/UploadForm.js`
  - `issue_tracker/jsx/IssueForm.js`
  - `issue_tracker/jsx/IssueTrackerBatchMode.js`
  - `media/jsx/uploadForm.js`
  - `genomic_browser/jsx/tabs_content/filemanager/uploadForm.js`
  - `modules/form/DateTimePartialElement.tsx`

- **Ensured error messages** are displayed based on the presence of an `errorMessage` prop, simplifying the error handling logic.

## Reason for Change

- **Simplifies error handling** by removing redundant `hasError` boolean prop.
- **Reduces complexity** in form components by relying on a single prop (`errorMessage`) for error states.
- **Improves consistency** across all form components in how errors are managed and displayed.

Please review the changes and let me know if any adjustments are needed.

### CODE REVIEW

**Review by maximemulder — APPROVED** [review]
Works on my machine. Nice cleaning PR.

It seems there are many approved Javascript PRs pending right now, this might create conflicts. I'll try to push to have the merged soon.

File: modules/document_repository/jsx/deleteCategoryForm.js [hunk 1]
```diff
@@ -106,9 +105,11 @@ class DeleteDocCategoryForm extends React.Component {
     );
   }
 
-  /** *******************************************************************************
+  /**
+   * *******************************************************************************
```
Comment (driusan) [inline modules/document_repository/jsx/deleteCategoryForm.js:109]: ?

Comment (maximemulder) [reply modules/document_repository/jsx/deleteCategoryForm.js:109]: @driusan I think this is an autoformatting change, it did the same thing when I used `--fix` for the indent linting PR (though I reverted it manually). So this is a problem with either the linter configuration or this code.

IMO not a problem to merge the PR with this (unimportant) change but do as you want.

Comment (driusan) [reply modules/document_repository/jsx/deleteCategoryForm.js:109]: I guess both the auto-corrected and the uncorrected format look kind of hideous in the code.

**Pull request merged**